components/brave_news/browser/topics_fetcher.h

```diff
@@ -0,0 +1,93 @@
+// Copyright (c) 2023 The Brave Authors. All rights reserved.
+// This Source Code Form is subject to the terms of the Mozilla Public
+// License, v. 2.0. If a copy of the MPL was not distributed with this file,
+// You can obtain one at https://mozilla.org/MPL/2.0/.
+
+#ifndef BRAVE_COMPONENTS_BRAVE_NEWS_BROWSER_TOPICS_FETCHER_H_
+#define BRAVE_COMPONENTS_BRAVE_NEWS_BROWSER_TOPICS_FETCHER_H_
+
+#include <string>
+#include <vector>
+
+#include "base/functional/callback_forward.h"
+#include "brave/components/api_request_helper/api_request_helper.h"
+#include "services/network/public/cpp/shared_url_loader_factory.h"
+
+namespace brave_news {
+
+constexpr char kTopicsEndpoint[] = "/news-topic-clustering/topics";
+constexpr char kTopicArticlesEndpoint[] = "/news-topic-clustering/topics_news";
+
+struct TopicArticle {
+  TopicArticle();
+  TopicArticle(const TopicArticle&);
+  TopicArticle(TopicArticle&&);
+  TopicArticle& operator=(const TopicArticle&);
+  ~TopicArticle();
+
+  std::string title;
+  std::string description;
+  GURL url;
+  GURL img;
+  std::string publisher_name;
+  base::Time publish_time;
+  double score;
```

Comment:
please always default-initialize POD-types in headers (applied to this and the struct below).